apps/dutchclock/app.js

```diff
@@ -0,0 +1,266 @@
+const storage = require("Storage");
+const locale = require('locale');
+const widget_utils = require('widget_utils');
+
+const SCREEN_WIDTH = g.getWidth();
+const SCREEN_HEIGHT = g.getHeight();
+
+const TOP_SPACING = 5;
+const WIDGETS_HEIGHT = 20;
+const DATETIME_SPACING_HEIGHT = 5;
+const TIME_HEIGHT = 10;
+const DATE_HEIGHT = 10;
+const BOTTOM_SPACING = 5;
+
+const TEXT_WIDTH = SCREEN_WIDTH - 2;
+
+const MINS_IN_HOUR = 60;
+const MINS_IN_DAY = 24 * MINS_IN_HOUR;
+
+const VARIANT_EXACT = 'exact';
+const VARIANT_APPROXIMATE = 'approximate';
+const VARIANT_HYBRID = 'hybrid';
+
+const DEFAULTS_FILE = "dutchclock.default.json"; 
+const SETTINGS_FILE = "dutchclock.json";
+
+// Load settings
+const settings = Object.assign(
+    storage.readJSON(DEFAULTS_FILE, true) || {},
+    storage.readJSON(SETTINGS_FILE, true) || {}
+);
+
+const maxFontSize = SCREEN_HEIGHT 
+  - TOP_SPACING
+  - (settings.showWidgets ? WIDGETS_HEIGHT : 0)
+  - (settings.showDate || settings.showTime ? DATETIME_SPACING_HEIGHT : 0)
+  - (settings.showDate ? DATE_HEIGHT : 0)
+  - (settings.showTime ? TIME_HEIGHT : 0);
+
+const X = SCREEN_WIDTH / 2;
+const Y = SCREEN_HEIGHT / 2
+  + TOP_SPACING / 2
+  + (settings.showWidgets ? WIDGETS_HEIGHT / 2 : 0)
+  - (settings.showDate || settings.showTime ? DATETIME_SPACING_HEIGHT / 2 : 0)
+  - (settings.showDate ? DATE_HEIGHT / 2 : 0)
+  - (settings.showTime ? TIME_HEIGHT / 2 : 0);
+  
+let date, mins;
+
+function initialize() {
+  // Tell Bangle this is a clock
+  Bangle.setUI("clock");
+  // Load widgets
+  Bangle.loadWidgets();
+
+  // draw immediately at first
+  tick();
+
+  // now check every second
+  let secondInterval = setInterval(tick, 1000);
+  // Stop updates when LCD is off, restart when on
+
+  Bangle.on('lcdPower',on=>{
+    if (secondInterval) clearInterval(secondInterval);
+    secondInterval = undefined;
+    if (on) {
+      secondInterval = setInterval(tick, 1000);
+      draw(); // draw immediately
+    }
+  });
+}
+
+function tick() {
+  date = new Date();
+  const m = (date.getHours() * MINS_IN_HOUR + date.getMinutes()) % MINS_IN_DAY;
+
+  if (m !== mins) {
+    mins = m;
+    draw();
+  }
+
+  if (!settings.showWidgets) {
+    widget_utils.hide();
+  }
```

Comment:
You don't need to hide them on every tick. If you tell them to hide they will stay hidden till you explicitly tell them to show again.

So if you follow my other comment I wrote now you can probably just delete this.